yb-voyager/cmd/importData.go

```diff
@@ -1206,6 +1223,33 @@ func extractCopyStmtForTable(table string, fileToSearchIn string) {
 	}
 }
 
+// get the parent table if its a partitioned table in yugabytedb
+// also covers the case when the partitions are further subpartitioned to multiple levels
+func getParentTable(table string, conn *pgx.Conn) string {
+	var parentTable string
+	query := fmt.Sprintf(`SELECT inhparent::pg_catalog.regclass
+	FROM pg_catalog.pg_class c JOIN pg_catalog.pg_inherits i ON c.oid = inhrelid
+	WHERE c.oid = '%s'::regclass::oid`, table)
+
+	var currentParentTable string
+	err := conn.QueryRow(context.Background(), query).Scan(&currentParentTable)
+	if err != pgx.ErrNoRows && err != nil {
+		utils.ErrExit("Error in querying parent tablename for table=%s: %v", table, err)
+	}
+
+	if len(currentParentTable) == 0 {
+		return ""
+	} else {
+		parentTable = getParentTable(currentParentTable, conn)
+		if len(parentTable) == 0 {
+			parentTable = currentParentTable
```

Comment:
Hmm, this recursion can be cut down by adding only one check of the type `len(table name)==0`, try exiting early by checking the length of the `table` argument as soon as you enter the function. 

P.S. You may need to alter the function return type and assign the "parent table" to a pointer declared outside the function, haven't given it much thought, but speaking from past experiences 🙂

Reply:
current function code is also similar. Just the base condition is at different point, not passing the empty tableName to the function, stopping it before that.
Its better not to use some global or outside variable and just make the function as separated out as possible.